services/py-api/src/utils.py

```diff
@@ -43,3 +45,37 @@ async def stop(self) -> None:
             raise RuntimeError("The AsyncClient has not been initialized!")
 
         await self._async_client.aclose()
+
+class JwtUtility:
+    class UserData(TypedDict):
+        id: int
+        name: str
+        email: str
+        is_admin: bool
+        email_verified: bool
+        team_id: int
+        created_at: str
+        updated_at: str
+         
+
+    @staticmethod
+    def encode(data: UserData) -> str:
+            return encode(data,key=environ["SECRET_KEY"])
+    
+    @staticmethod
+    def decode(token: str) -> Dict[str, Any]:
+     try:
+        decoded_payload=jwt.decode(token, SECRET_KEY, algorithms=["HS256"])
+        return decoded_payload
+     except ExpiredSignatureError:
+            print("The token has expired.")
+            return {"error": "TokenExpired"}
+     except InvalidSignatureError:
+            print("The token has an invalid signature.")
+            return {"error": "InvalidSignature"}
+     except DecodeError:
+            print("There was an error decoding the token.")
+            return {"error": "DecodeError"}
+    
+
+print(JwtUtility.encode({"str":"fdsafsad"}))
```

Comment:
I see that you are trying to test it here. but it won't really work if you just start the backend server. Since the backend server is and HTTP server it will only trigger methods based on the requests in gets in its endpoints. Meaning, you have to go to py-api/server/routes/utility_routes.py and add a route that triggers this function 

```Python
from src.utils.py  import JwtUtility
@utility_router.get("/testjwt", responses={200: {"model": PongResponse}, 503: {"model": ErrResponse}})
async def testJWT() -> Any:
    return await JwtUtility.encode({"str":"fdsafsad"})
```

Open postman and do a GET request in https://localhost:8080/testjwt and it should trigger your function and you will be able to see the message in your console.

Reply:
No need for this, I suggest @plamenapavlova write "integration-test" for this as writing a unit test is not going to do much work for us. 
What the intergration test should do:

1. Encode some info and then try to decode it  -> it should return success
2. Encode invalid info and check if returns an error
3. Decode an expired jwt token (you could have it hardcoded in a const)
4. Decode a jwt token which is signed with a different secret (you could have it the test-secret in a const)
5. Decode a malformed JWT